clang/lib/Basic/Targets/LoongArch.cpp

```diff
@@ -388,6 +388,76 @@ bool LoongArchTargetInfo::handleTargetFeatures(
   return true;
 }
 
+enum class AttrFeatureKind { Arch, Tune, NoFeature, Feature, Invalid };
+
+static std::pair<AttrFeatureKind, llvm::StringRef>
+getAttrFeatureTypeAndValue(llvm::StringRef AttrFeature) {
+  if (auto Split = AttrFeature.split("="); !Split.second.empty()) {
+    if (Split.first.trim() == "arch")
+      return {AttrFeatureKind::Arch, Split.second.trim()};
+    if (Split.first.trim() == "tune")
+      return {AttrFeatureKind::Tune, Split.second.trim()};
+  }
+  if (AttrFeature.starts_with("no-"))
+    return {AttrFeatureKind::NoFeature, AttrFeature.drop_front(3).trim()};
+  return {AttrFeatureKind::Feature, AttrFeature.trim()};
```

Comment:
The caller has already  performed `trim()`, so it's unnecessary here, just like `starts_with()` above doesn't use it either.

```suggestion
  return {AttrFeatureKind::Feature, AttrFeature};
```